pkg/operator/ceph/pool/controller.go

```diff
@@ -160,17 +160,19 @@ func (r *ReconcileCephBlockPool) reconcile(request reconcile.Request) (reconcile
 		return opcontroller.ImmediateRetryResult, errors.Wrap(err, "failed to get CephBlockPool")
 	}
 
+	// The CR was just created, initializing status fields
```

Comment:
Re-ordering it will not provide any impact?

Reply:
I'm not sure it really matters if the status is initially updated before or after the finalizer is added, but did you see a need to change the order? If not, I'd suggest we keep the initial status update after the finalizer is added so we don't risk a side effect.

Reply:
Oh are you changing this order so the finalizer update will apply the observed generation? There are a couple issues with this approach:
- This will only apply the generation during the first reconcile. All future reconciles will skip adding the finalizer since it already exists.
- We want to apply the observed generation at the end of the reconcile instead of the beginning.

Reply:
okay